libcxx/test/std/utilities/optional/optional.iterator/borrowed_range.compile.pass.cpp

```diff
@@ -0,0 +1,53 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: std-at-least-c++26
+
+// <optional>
+
+// template <class T> class optional<T&>::iterator;
+// template <class T> class optional<T&>::const_iterator;
+// template <class T>
+// constexpr bool ranges::enable_borrowed_range<optional<T&>> = true;
+
+#include <cassert>
+#include <optional>
+#include <ranges>
+
+template <typename T>
+constexpr void enable_borrowed_range() {
+  static_assert(std::ranges::enable_borrowed_range<std::optional<T&>>);
+}
+
+template <typename T>
+constexpr void borrowed_range() {
+  static_assert(std::ranges::range<std::optional<T&>> == std::ranges::borrowed_range<std::optional<T&>>);
+}
```

Comment:
Would it be simpler to put these two `static_assert`s in the same function template? Also, I think it's better to drop `constexpr` here as these function templates are only instantiated and function calls are not evaluated.